src/python/pants/backend/python/tasks/resolve_requirements_task_base.py

```diff
@@ -51,13 +51,17 @@ def resolve_requirements(self, interpreter, req_libs):
       else:
         target_set_id = 'no_targets'
 
+      # We need to ensure that we are resolving for only the current platform if we are
+      # including local python dist targets that have native extensions.
+      maybe_platforms = ['current'] if self.tgt_closure_has_native_sources() else None
```

Comment:
ditto. tho this code path is slightly less important than the one above (because presumably this is the path called for things like `run` or `test` which execute locally), a user would probably expect e.g. an unsatisfiable transitive dep for e.g. `linux-x86_64` to manifest as a failure in `test` locally on an OSX machine - which won't happen in this case.